aten/src/ATen/native/Linear.cpp

```diff
@@ -201,8 +201,8 @@ Tensor einsum(std::string eqn, TensorList tensors) {
         }
       } else {                                          // a letter (hopefully)
         TORCH_CHECK((ell_char_count == 0) || (ell_char_count == 3), "'.' must only occur in ellipsis, operand ", operand);
-        TORCH_CHECK(('a' <= c) && (c <= 'z'), "only lowercase letters a-z allowed as indices");
-        int64_t letter_num = c-'a';                     // letter_num  = position in letter_mapping
+        TORCH_CHECK(('A' <= c) && (c <= 'z'), "only letters [A-Z,a-z] allowed as indices");
```

Comment:
Oh, it's not sound to treat A-Z and a-z as one contiguous range; there is some punctuation in the middle which you need to reject.